### DISCUSSION TITLE
Using Redux Persist

### DISCUSSION THREAD

**Discussion by simplenotezy** [vercel/next.js#31884] [Help]
Did anybody mange to get Redux Persist working on Next.js?

- https://github.com/rt2zz/redux-persist/issues/1008
- https://github.com/rt2zz/redux-persist/issues/1187
- https://github.com/rt2zz/redux-persist/issues/598

I have managed to find this example from Next.js:

https://github.com/vercel/next.js/tree/canary/examples/with-redux-persist

But they have a SEO issue, in that the page will display "loading..." if client does not have javascript enabled.

**Comment by simplenotezy** [comment]
So far I have tried without using `PersistGate` and for our use case, it actually works, and store is hydrated instantly in my tests.

For anybody who's having pain with this right now, as I've had a lot, below is an example of my setup, for you to see:

**store.js**

	import { configureStore } from '@reduxjs/toolkit';
	import { setupListeners } from '@reduxjs/toolkit/dist/query';
	import storage from 'redux-persist/lib/storage';
	import { combineReducers } from 'redux';
	import { persistReducer } from 'redux-persist';
	import auth from '@features/Auth/store';
	import { authApi } from '@features/Auth/services/auth';
	import persistStore from 'redux-persist/lib/persistStore';
	import app from './app';

	const reducers = combineReducers({
		app,
		auth,
		[authApi.reducerPath]: authApi.reducer,
	});

	const persistConfig = {
		key: 'root',
		storage,
		version: 1,

		// the stores you wish to persist in localstorage
		whitelist: [
			'app',
			'auth',
			authApi.reducerPath,
		],
	};

	const persistedReducer = persistReducer(persistConfig, reducers);

	const store = configureStore({
		reducer: persistedReducer,

		// Adding the api middleware enables caching, invalidation, polling,
		// and other useful features of `rtk-query`.
		// middleware: [rtkQueryFailed, authApi.middleware, authApi.middleware, userApi.middleware],
		middleware: (getDefaultMiddleware) =>
			getDefaultMiddleware({ serializableCheck: false })
				.concat(authApi.middleware),
	});

	setupListeners(store.dispatch);

	export const persistor = persistStore(store, {}, () => {
		persistor.persist();
	});

	export default store;

And in your `_app.js`, simply wrap ReduxProvider without using PersistGate:

**_app.js**
```
import store from 'store.js';

[...]

return (
    <ReduxProvider store={store}>
         .....
```

Hope it can help you!

**Reply by ibraheemosule** [reply]
I used this method you proposed, and while the data actually persists (checked my local storage and it was there), the persisted data doesnt get updated to the rtk query data
when i refresh my page and log data from use mutation rtk hook, i get undefined returned even though the persisted data exists in the local storage

**Reply by valentinesamuel** [reply]
Hello @ibraheemosule , did you find a solution to the problem?. Because i am having exactly the same problem

**Reply by ibraheemosule** [reply]
Yes Bros, I did but I can't remember the solution until i check the project again
Please, reach out on twitter to me on @ibraheemosule
